src/test/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJobTest.java

```diff
@@ -49,21 +55,27 @@ public void execute_should_send_http_request_specified_in_job_details() {
         given(context.getJobDetail())
             .willReturn(newJob(HttpCallJob.class).withIdentity("id", "group").build());
 
+        Map<String, String> headers = new HashMap<>();
+        headers.put("X-Custom-Header", "anything");
+        headers.put("ServiceAuthorization", "some-token");
+
         given(actionExtractor.extract(context))
             .willReturn(new HttpAction(
                 "http://localhost:8080/hello-world",
                 HttpMethod.POST,
-                ImmutableMap.of("ServiceAuthorization", "some-token"),
+                ImmutableMap.copyOf(headers),
```

Comment:
I think we misunderstood each other. I guess what Rafał, seeing that you moved from ImmutableMap to HashMap, wanted to say that it's still possible to do the same with ImmutableMap:

```
ImmutableMap.of(
    "X-Custom-Header", "anything",
    "ServiceAuthorization", "some-token"
)
```

This line was ok as it was (without `ImmutableMap.copyOf`), and I suppose the way headers Map is defined is also fine (although ImmutableMap is even better).

Reply:
I've refactored the tests.